src/java/org/apache/cassandra/index/sai/memory/TrieMemtableIndex.java

```diff
@@ -117,7 +118,7 @@ public ByteBuffer getMinTerm()
         return Arrays.stream(rangeIndexes)
                      .map(MemoryIndex::getMinTerm)
                      .filter(Objects::nonNull)
-                     .min(Comparator.comparing(identity(), validator))
+                     .reduce((a, b) -> TypeUtil.min(a, b, validator))
```

Comment:
I wanted to understand why this fixed it, here are my results. The change is very subtle. It fixes the failing tests because `TypeUtil#compare` has some special logic for specific types before calling `validator#compare`:

```
    public static int compare(ByteBuffer b1, ByteBuffer b2, AbstractType<?> type)
    {
        if (isInetAddress(type))
            return compareInet(b1, b2);
        // BigInteger values, frozen types and composite types (map entries) use compareUnsigned to maintain
        // a consistent order between the in-memory index and the on-disk index.
        else if (isBigInteger(type) || isBigDecimal(type) || isCompositeOrFrozen(type))
            return FastByteOperations.compareUnsigned(b1, b2);

        return type.compare(b1, b2 );
    }
```

This makes me question why we don't have the correct `type.compare` definitions in the respective classes--likely something that has historical reasons.

Reply:
I guess the comment explains why we do this, but it seems error prone, as we found out in this PR.

Reply:
I am curious why these types aren't stored in their byte-ordered translation. Don't we want to ever perform range queries e.g. on decimals?

Reply:
> I am curious why these types aren't stored in their byte-ordered translation

I forgot the context about big-int and big decimal.

> Don't we want to ever perform range queries e.g. on decimals?

range queries are supported on decimals with kdtree